### DISCUSSION TITLE
The best use of useFieldArray with a database.

### DISCUSSION THREAD

**Discussion by mwlai288** [react-hook-form/react-hook-form#4904] [Q&A]
At work I am retrieving data and displaying the information in forms. I am also making use of `useFieldArray` for the user to submit new data.  The way the backend was set up is that there will be a submit for all the information at once, the new data and old data. How will I be able to have unique `name`s for the inputs from `useFieldArray` and the inputs that have been created from the database or even displaying the information from the database in `useFieldArray`?

**Comment by bluebill1049** [comment]
you can store that unique name as a hidden input?

**Reply by mwlai288** [reply]
there is another thing. when I append a new field to the fields array when there are already existing data shown, it'll add the number of inputs already existing plus a new one. by that I mean, say that there are 3 inputs already created from the information in the database, when I want to add one new field, it'll add 3 inputs that arent registered plus the new input that gets focused and works as regular input. I really hope this makes sense. It's hard to explain, 
I can try and make a code sandbox to illustrate my issue better.

By hidden input you mean the inputs with the saved data or the newly created inputs?

**Reply by bluebill1049** [reply]
Yea, I think a simple codesandbox would be helpful.

**Reply by mwlai288** [reply]
This is the closest I came to reproducing my issue. When I click append I get additional inputs before the input I really need. The inputs have the same name because I need to resubmit the existing data again when I create a new entry.
I'm still working on version 6*

https://codesandbox.io/s/usefieldarray-help-qjx7q

**Reply by bluebill1049** [reply]
Thanks @mwlai288 what's the steps to reproduce?

**Reply by mwlai288** [reply]
Press the append button.
I'm just looking to get one input and not 5.

**Reply by bluebill1049** [reply]
oh, I see what do you mean because there are 4 inputs already got registered. if you just want to add one then I don't think you need `useFieldArray`, you can create a separate form just to register the new field and update the value.

**Reply by mwlai288** [reply]
The reason I wanted to use `useFieldArray` was for the user to fill multiple entries at first then add more if they want after the initial submit. That was one way we thought of handling it but I think since the way the backend was set up, for the existing data to be resubmitted with new data makes it tough to handle.

**Reply by imadeelnazar** [reply]
Hey @mwlai288 Any Luck with Filling out multiple entries and then trying append or delete previous, or any resource you know to handle that?
